keylime-ansible-gcp/README.md

```diff
@@ -0,0 +1,53 @@
+# Ansible Keylime for Google Cloud
+Ansible role to deploy a Fedora 35 instance on the Google Cloud Platform with [Keylime](https://github.com/keylime/keylime) and the [rust agent](https://github.com/keylime/rust-keylime) against a Virtualized TPM.
+
+Contributions are welcome, should anyone wish to have this role provision other Linux distributions.
+
+For details on using Keylime, please consult the
+[project documentation](https://keylime-docs.readthedocs.io/en/latest/)
+
+For details on the Rust agent, please consult the [repository](https://github.com/keylime/rust-keylime).
+
+## Configuration
+1.  Install dependecies for the Ansible - GCP module 
+`$ pip3 install requests google-auth`
+2. [Create a GCP service account](https://developers.google.com/identity/protocols/oauth2/service-account#creatinganaccount) for ansible's use.
+3. [Create and download keys linked to this service account](https://support.google.com/cloud/answer/6158849?hl=en&ref_topic=6262490#serviceaccounts&zippy=%2Cservice-accounts)
+4. Add path to ssh key to ansible config (/etc/ansible/ansible.conf) \
+Note: the ssh private key is in the downloaded file. Extract it, place it in its own file, set adequate permissions, and add the path to this newly created file to the ansible config. \
+Example:\
+[defaults] \
+private_key_file = /home/user/my_key
+
+5. Set environment variables \
+`$ export GCP_PROJECT="<name of GCP project>"` \
+`$ export GCP_CRED_KIND="serviceaccount"`\
+`$ export GCP_CRED_FILE="<path to your service account key file>"` \
+`$ export GCP_ZONE="<zone for GCP instance>"` \
+`$ export GCP_REGION="<region for GCP instance>"` 
+
+## Usage
+Run the playbook to create and set up an instance. 
+
+```bash
+ansible-playbook playbook.yml
```

Comment:
I got this error:
```
[lily@localhost keylime-ansible-gcp]$ ansible-playbook playbook.yml
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
ERROR! couldn't resolve module/action 'google.cloud.gcp_compute_disk'. This often indicates a misspelling, missing collection, or incorrect module path.

The error appears to be in '/home/lily/Repos/keylime-cloud-environments/keylime-ansible-gcp/roles/create-gcp/tasks/gcp.yml': line 1, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

- name: Create disk mapped from Fedora image
  ^ here
```

Reply:
[This](https://stackoverflow.com/questions/38328401/warning-provided-hosts-list-is-empty-only-localhost-is-available-when-invent) looks like it may be the problem but I'm not sure what to enter for hostname / username, do you know?